File: python/1302. Deepest Leaves Sum.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def deepestLeavesSum(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        res = 0
        que = [root]
        while que:
            sz = len(que)
            res = 0
            while sz:
                sz -= 1
                root = que.pop(0)
                if root.left:
                    que.append(root.left)
                if root.right:
                    que.append(root.right)
                res += root.val
        return res
            

File: go/src/leetcode/1302/1302. Deepest Leaves Sum.go
/**
@author ZhengHao Lou
Date    2022/08/17
*/
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func deepestLeavesSum(root *TreeNode) int {
	var ans int
	que := []*TreeNode{root}
	for len(que) != 0 {
		var s int
		for l := len(que); l > 0; l-- {
			x := que[0]
			que = que[1:]
			s += x.Val
			if x.Left != nil {
				que = append(que, x.Left)
			}
			if x.Right != nil {
				que = append(que, x.Right)
			}
		}
		ans = s
	}
	return ans
}
